src/features/user/slice.ts

```diff
@@ -0,0 +1,31 @@
+// TODO: State needs to be updated if the user's info changes or the session
+// gets invalidated
+import { PayloadAction, createSlice } from '@reduxjs/toolkit'
+
+import { UserProfile } from '../../services/userService'
+import { UserState } from './types'
+
+const initialState: UserState = {
+  isLoggedIn: isLoggedIn(),
+}
+
+const userSlice = createSlice({
+  name: 'user',
+  initialState,
+  reducers: {
+    updateIsLoggedIn: (state) => {
+      state.isLoggedIn = isLoggedIn()
+    },
+    setUserProfile: (state, action: PayloadAction<UserProfile | undefined>) => {
+      state.userProfile = action.payload
+    },
+  },
+})
+
+function isLoggedIn() {
+  return document.cookie.includes('is-logged-in=true')
+}
```

Comment:
Regarding `APP_LOGGED_IN_INIT`, understand that it would be good to centralise all app init in one location.

However, I was thinking that it would be useful to have a location to run all things initialisation related that should only be done for a logged in user. Stuff like:
- Fetching user profile
- Fetching user question attempt history
- Fetching user matching status (so if a user is already in a collaboration, they should be redirected to the collaboration page immediately)

Since they won't all be handled by the same service, I feel this will likely help in avoiding the issue of each such Redux Saga event having to do a `document.cookie('is-logged-in'=true)` call.